2002-08-26  Jan Hubicka  <jh@suse.cz>
	
	* tc-i386.c (tc_i386_fix_adjustable): Recognize GOTPCREL properly.

--- binutils-2.13.90.0.10/gas/config/tc-i386.c.x86_64-gotpcrel	2002-11-15 15:16:42.000000000 +0100
+++ binutils-2.13.90.0.10/gas/config/tc-i386.c	2002-11-15 15:22:40.000000000 +0100
@@ -1220,6 +1220,12 @@ tc_i386_fix_adjustable (fixP)
       && fixP->fx_pcrel)
     return 0;
 
+  /* The x86_64 GOTPCREL are represented as 32bit PCrel relocations
+     and changed later by validate_fix.  */
+  if (GOT_symbol && fixP->fx_subsy == GOT_symbol
+      && fixP->fx_r_type == BFD_RELOC_32_PCREL)
+    return 0;
+
   /* adjust_reloc_syms doesn't know about the GOT.  */
   if (fixP->fx_r_type == BFD_RELOC_386_GOTOFF
       || fixP->fx_r_type == BFD_RELOC_386_PLT32
